src/gluonts/mx/trainer/_base.py

```diff
@@ -344,10 +393,17 @@ def loop(
                             best_epoch_info["params_path"], self.ctx
                         )
 
-                logging.info("Computing averaged parameters.")
-                averaged_params_path = self.avg_strategy.apply(gluonts_temp)
+                if isinstance(self.avg_strategy, AveragingStrategy):
+                    logging.info("Computing averaged parameters.")
+                    averaged_params_path = self.avg_strategy.apply(
+                        gluonts_temp
+                    )
+
+                    logging.info("Loading averaged parameters.")
+                    net.load_parameters(averaged_params_path, self.ctx)
 
-                logging.info("Loading averaged parameters.")
-                net.load_parameters(averaged_params_path, self.ctx)
+                if isinstance(self.avg_strategy, IterationAveragingStrategy):
+                    logging.info("Loading averaged parameters.")
+                    self.avg_strategy.load_averaged_model(net)
```

Comment:
My original impression was that we could combine the existing epoch-averaging approaches with the new iteration-averaging but it seems that this is not the case. Then, we could refactor the classes as follows: `AveragingStrategy` with subclasses `EpochAveragingStrategy` (under which `SelectNBestMean` etc. will be) and `IterationAveragingStrategy` under which the new methods will be.

Reply:
Though they look related semantically, they actually belong to different categories of approaches, and they use different APIs. I think adding another abstract class will make it more complicated and confusing. We also made similar problems when we designed GluonNLP and got some bad feedbacks about it. So now we are refactoring our APIs to make things clearer and easier for users to understand and develop their own functionalities.